### DISCUSSION TITLE
There are too many request parameter fields and cannot be filtered.

### DISCUSSION THREAD

**Discussion by twricu** [TongchengOpenSource/smart-doc#856] [Q&A] [answered]
For query interfaces such as paging and lists, I will use a `DTO` object to inherit the original entity class, and add some additional fields to the `DTO` object for extended reception.

However, because `@RequestBody xxxDTO` is an object, **smart-doc** will ignore the `@param` annotation in the document comment, resulting in a lot of fields in the generated request body parameter document that do not need to be passed, and The `@ignoreParams` annotation cannot filter the fields in the object that I do not want to display. Can I only use the `@apiNote` annotation to manually write or redefine an entity class?

Is there a better way to achieve this, or provide a separate annotation similar to a whitelist for the request type parameters?

**Comment by shalousun** [comment, marked as answer]
Smart-Doc supports using JSR validation rules to standardize fields. Fields annotated with the `@Null` annotation will not be displayed in the documentation.

**Reply by shalousun** [reply]
When raising a discussion next time, please use an AI translation tool to translate it into English, as we are an international open-source community.

**Reply by twricu** [reply]
> When raising a discussion next time, please use an AI translation tool to translate it into English, as we are an international open-source community.

Thanks for the answer, it has been translated into English.
